File: code_challenge_solution/c#/168 Reverse a string.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        Console.Write("Enter a string: ");
        string input = Console.ReadLine();

        string reversed = ReverseString(input);

        Console.WriteLine("Reversed string: " + reversed);
    }

    static string ReverseString(string str)
    {
        char[] charArray = str.ToCharArray();
        Array.Reverse(charArray);
        return new string(charArray);
    }
}

File: code_challenge_solution/java/168 Reverse a string.java

import java.util.Scanner;

public class ReverseString {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        // Input string from user
        System.out.println("Enter a string: ");
        String original = sc.nextLine();
        
        // Reverse the string
        String reversed = reverseString(original);
        
        // Print the reversed string
        System.out.println("Reversed string: " + reversed);
        
        sc.close();
    }
    
    public static String reverseString(String original) {
        StringBuilder sb = new StringBuilder(original);
        sb.reverse();
        return sb.toString();
    }
}
